daphne/src/roles.rs

```diff
@@ -497,16 +509,15 @@ where
 
         // Fetch reports and run an aggregation job for each task.
         for (task_id, reports) in self.get_reports(selector).await?.into_iter() {
-            let task_config = self
-                .get_task_config_for(Cow::Owned(task_id.clone()))
-                .await?
-                .ok_or(DapAbort::UnrecognizedTask)?;
-
             for (part_batch_sel, reports) in reports.into_iter() {
                 // TODO Consider splitting reports into smaller chunks.
                 // TODO Consider handling tasks in parallel.
                 telem.reports_processed += reports.len() as u64;
                 if !reports.is_empty() {
+                    let task_config = self
+                        .get_task_config_for(Cow::Owned(task_id.clone()))
+                        .await?
+                        .ok_or(DapAbort::UnrecognizedTask)?;
```

Comment:
What's the reason for moving this task_config lookup into the inner for-loop?

Reply:
Good catch; this was an artifact of a change I undid, but not fully.  Originally I was calling get_task_config_considering_taskprov() here, but it was too late so I moved the call to the right earlier place but forgot to undo this change.  I will fix it.  (The way things are now, the original code works fine as taskprov config instantiation will already have happened and the normal get_task_config_for() will find the task.)